lib/client.js

```diff
@@ -369,6 +369,13 @@ Client.prototype.query = function (config, values, callback) {
     }
   }
 
+  if (this._ending) {
+    if (query.callback) {
+      query.callback(new Error('Could not accept query, because connection has ended'))
+    }
+    return
```

Comment:
The non-callback case should return a rejected Promise. Otherwise `await client.query(...)` would resolve to `undefined`. I think all that needs to be changed is the return clause on L376 to:

```
return result
```

I haven't actually tried out the patch but that jumped out at me so figured to comment now if you're going to pursue this further. Could include both styles in a test as well (i.e. with callback and without).